sdk/core/abort-controller/src/AbortController.ts

```diff
@@ -58,8 +58,34 @@ export class AbortError extends Error {
 export class AbortController {
   private _signal: AbortSignal;
 
-  constructor() {
+  /**
+   * @param {AbortSignal} [parentSignal] The AbortSignal that will signal aborted on the AbortSignal associated with this controller.
+   * @constructor
+   */
+  constructor(parentSignals?: AbortSignalLike[]);
+  constructor(...parentSignals: AbortSignalLike[]);
+  constructor(parentSignals?: any) {
     this._signal = new AbortSignal();
+
+    if (!parentSignals) {
+      return;
+    }
+    // coerce parentSignals into an array
+    if (!Array.isArray(parentSignals)) {
+      parentSignals = arguments;
+    }
+    for (const parentSignal of parentSignals) {
+      // if the parent signal has already had abort() called,
+      // then call abort on this signal as well.
+      if (parentSignal.aborted) {
+        this.abort();
+      } else {
+        // when the parent signal aborts, this signal should as well.
+        parentSignal.addEventListener("abort", () => {
+          this.abort();
+        });
```

Comment:
Should we have a way to "close" an abort signal so that it can remove all the listeners? This is a separate conversation and need not block this PR.

Reply:
When would this get used?

Reply:
We could have the `abortSignal` method remove listeners from the `listenersMap` after it invokes all of them if we're worried about listeners not getting garbage collected. The listenersMap is a `WeakMap` though so I don't think we have to remove the references ourselves.

Reply:
only if you want the listeners gone while the signal itself is still held somewhere (i.e. stashed away inside a client or something). But since it's a weakmap, removing any references to the signal should have the same memory effects as clearing the listeners.